src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -1685,9 +1684,11 @@ public static Builder builder() {
    * @throws DockerCertificateException if we could not build a DockerCertificates object
    */
   public static Builder fromEnv() throws DockerCertificateException {
-    final String endpoint = fromNullable(getenv("DOCKER_HOST")).or(defaultEndpoint());
+    final DockerUtils dockerUtils = new DockerUtils();
+    final String endpoint = fromNullable(getenv("DOCKER_HOST"))
```

Comment:
Ah good point

On Thursday, May 19, 2016, Flavin <notifications@github.com> wrote:

> In src/main/java/com/spotify/docker/client/DefaultDockerClient.java
> <https://github.com/spotify/docker-client/pull/441#discussion_r63953389>:
>
> > @@ -1685,9 +1684,11 @@ public static Builder builder() {
> >     * @throws DockerCertificateException if we could not build a DockerCertificates object
> >     */
> >    public static Builder fromEnv() throws DockerCertificateException {
> > -    final String endpoint = fromNullable(getenv("DOCKER_HOST")).or(defaultEndpoint());
> > +    final DockerUtils dockerUtils = new DockerUtils();
> > +    final String endpoint = fromNullable(getenv("DOCKER_HOST"))
>
> This exact series of method calls—
> fromNullable(getenv("DOCKER_HOST")).or(dockerUtils.defaultDockerEndpoint());—is
> used multiple times in different places. It seems to me this would be
> better as a single method in DockerUtils, something like getHost().
>
> Same with
> fromNullable(getenv("DOCKER_CERT_PATH")).or(dockerUtils.defaultCertPath())
> .
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/spotify/docker-client/pull/441/files/ac2c911da1dc2013c62a4f2950d13ae9857742da#r63953389>
>


-- 
David Xia
Software Engineer
dxia@spotify.com